tests/test.js

```diff
@@ -369,3 +390,46 @@ exports['should expose `getUrl` method for wrapped signal'] = function (test) {
   test.done();
 
 };
+
+exports['should match regexp'] = function (test) {
+
+  global.location.href = '/madonna-tickets-any';
+
+  var controller = createController();
+  controller.signal('test', [
+    function (input) {
+      test.ok(true);
+      // console.log(input);
+    }
+  ]);
+
+  Router(controller, {
+    '/:artist(\\w+)-tickets(.*)': 'test'
+  }).trigger();
+
+  test.done();
+
+};
+
+exports['should redirect'] = function (test) {
```

Comment:
can't figure how to test replace part. we can't bring real `addressbar` to tests easily. It will require history api mocking.
we can update addressbar stub to support object in setter and check if it had `replace: true` property